libfreerdp/CMakeLists.txt

```diff
@@ -248,23 +248,15 @@ if(FAAC_FOUND)
 endif()
 
 if (WITH_AAD)
-    if (NOT cJSON_FOUND)
-        find_package(PkgConfig REQUIRED)
-        pkg_check_modules(CJSON libcjson)
-    endif()
-    if (NOT CJSON_LIBRARIES OR NOT CJSON_INCLUDE_DIRS)
-        find_path(CJSON_INCLUDE_DIRS
-            NAMES cjson/cJSON.h
-            REQUIRED
-        )
-        find_library(CJSON_LIBRARIES
-            NAMES cjson
-            REQUIRED
-        )
+    if (WITH_JSONC)
```

Comment:
this should still search for the library (with `REQUIRED` keyword).
the detection before is used to determine the default value of the `WITH_AAD` option, but here we enforce the requirements.

Reply:
Isn't the `message(FATAL_ERROR "AAD support requested but cJSON not found")` enough?